addons/web/static/src/search/control_panel/control_panel.xml

```diff
@@ -28,13 +28,13 @@
                                  <span class="visually-hidden">Toggle Dropdown</span>
                             </button>
                             <ul class="dropdown-menu o_control_panel_collapsed_create">
-                                <t t-slot="layout-buttons"/>
+                                <t t-if="!env.inDialog" t-slot="layout-buttons"/>
```

Comment:
this seems to radical. You don't know how the parent of the control panel use the slots

Reply:
agreed, the component defiining the slot should define that.
something like 
```xml 
<t-name listcontroller>
<ControlPanel >
    <t t-set-slot="layout-button" shouldDisplay="!env.inDialog" />
</ControlPanel>
</t>


<t-name ControlPanel>

           <t t-if="props.slots["layout-buttons"].shouldDisplay" t-slot="layout-buttons" />

</t>

```

Reply:
Can't we set a `t-if`of the `t-set-slot` ? Not a big fan of the `shouldDisplay` thing

Reply:
layout-buttons has not default content so a t-if on t-set-slot in the parent is enough I think